x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MigrateAction.java

```diff
@@ -102,6 +106,23 @@ public boolean isSafeAction() {
         }
     }
 
+    /**
+     * Based on the provided target tier it will return a comma separated list of preferred tiers.
+     * ie. if `data_cold` is the target tier, it will return `data_cold,data_warm,data_hot`
+     * This is usually used in conjunction with {@link DataTierAllocationDecider#INDEX_ROUTING_PREFER_SETTING}
+     */
+    static String getPreferredTiersConfiguration(String targetTier) {
+        int indexOfTargetTier = COLD_TO_HOT_TIERS.indexOf(targetTier);
+        if(indexOfTargetTier == -1) {
+            throw new IllegalArgumentException("invalid data tier [" + targetTier + "]");
+        }
+        String preferredTiers = COLD_TO_HOT_TIERS.stream().skip(indexOfTargetTier).collect(Collectors.joining(","));
+        if (Strings.hasText(preferredTiers) == false) {
```

Comment:
If we keep this, we could reverse the double negative with
```suggestion
        if (Strings.isEmpty(preferredTiers)) {
```

But what I'm trying to understand is how this would actually be empty, even in the hot phase the stream skip above would at least return `"data_hot"`, and we check that the index is not -1 above, so I'm not sure how it actually get into this state?

Reply:
this is my paranoia, you're right, I'll drop it